hugo/content/docs/langium-overview.md

```diff
@@ -0,0 +1,103 @@
+- [Editor support](#editing-support)
+
+With that in mind, Langium provides out-of-the-box solutions for these problems, with the ability to fine-tune every part of it to fit your domain requirements.
+
+---
+
+## Language Parsing
+
+Programming languages and domain specific languages (DSL) cannot be parsed using simple regular expressions (RegExp). Instead they require a more sophisticated parsing strategy. To define a custom language, you'll interact with a high level representation of your context-free grammar using the [Langium grammar language](../grammar-language), in a similar fashion to EBNF.
+
+Based on the grammar, Langium is then able to construct a parser which transforms an input string into a semantic model representation.
+
+## Semantic Models
+
+Langium grammars are not only used to parse your language, but also generate a semantic model for your Language as TypeScript interfaces, which automatically match the parsed abstract syntax tree. The following language snippet:
+
+```ts
+Person:
+    'person' // keyword 
+    name=ID // semantic assignment
+;
+```
+
+Parses a simple object `{ name: 'John' }`. To interact with this object in a type safe manner, the `langium-cli` tool generates TypeScript type definitions from its parser rule. The `Person` parser rule generates the following interface:
+
+```ts
+interface Person {
+    name: String
+}
+```
+
+These interfaces allow you to safely traverse your abstract syntax tree. In case your grammar changes, they will also notify you of any breaking changes which might require you to change your domain logic.
+
+## Cross References and Linking
+
+To express any kind of relationship between elements in your language, you will need to **reference** them.
+The process of resolving these references, i.e. identifying what element of your language hides behind a certain name, is called _linking_.
+Performing the linking process in a deterministic manner with a lot of objects in your project requires sound linking design.
+
+Langium accomplishes this feat by using the concept of 'scoping'. You likely know scopes from programming, where some variables are only available from certain scopes:
+
+```ts
+let x = 42;
+x = 3; // References the `x` defined in the previous line
+```
+
+```ts
+if (something) {
+    let y = 42;
+}
+y = 3; // Cannot link, `y` isn't in any of the available scopes
+```
+
+The same occurs in Langium. To enable more complex scoping behavior, you can create a custom domain scoping. For example, common object-oriented languages need a more involved scoping mechanism to resolve references to fields and methods of a class:
+
+```ts
+class X {
+    y(): void { ... }
+}
+
+const instance = new X(); // Symbol `X` is in the local scope
+instance.y(); // Symbol `y` exists in the scope of the `X` class
+```
+
+Once your domain specific scoping rules have been defined, Langium will take care of linking by itself, reporting any errors appearing 
```

Comment:
missing period at the end, plus can probably drop 'appearing' and just say `reporting any errors.`